src/Tasks/SystemState.cs

```diff
@@ -94,6 +102,11 @@ internal sealed class SystemState : StateFileBase, ISerializable
         /// </summary>
         private DirectoryExists directoryExists;
 
+        /// <summary>
+        /// Cached delegate.
+        /// </summary>
+        private DirectoryGetFiles getFiles;
```

Comment:
minor: naming consistency.  Elsewhere, instance fields are prefixed with "_"; thus, ```_getFiles".

Reply:
In this particular class 'SystemState' are all private fields without _